src/memory.rs

```diff
@@ -125,6 +143,11 @@ impl Memory {
         self.alloc_map.get_mut(&id).ok_or(EvalError::DanglingPointerDeref)
```

Comment:
Currently `get` and `get_mut` would return `Err(EvalError::DanglingPointerDeref)` given a function pointer's `AllocId`. They should probably check `self.functions` in the `None` case and only return `DanglingPointerDeref` if `self.functions.get_mut(&id)` is also `None`.